src/containers/main/Airdrop/AirdropGiftTracker/hooks/useAirdropAuth.ts

```diff
@@ -0,0 +1,62 @@
+import { useCallback } from 'react';
+import { open } from '@tauri-apps/plugin-shell';
+import { v4 as uuidv4 } from 'uuid';
+import { useAirdropStore } from '@app/store/useAirdropStore';
+import { useMiningStore } from '@app/store/useMiningStore';
+
+export const useAirdropAuth = () => {
+    const { backendInMemoryConfig, authUuid, setAuthUuid, setAirdropTokens } = useAirdropStore();
+    const restartMining = useMiningStore((s) => s.restartMining);
+
+    const handleAuth = useCallback(
+        (code?: string) => {
+            const token = uuidv4();
+            if (backendInMemoryConfig?.airdropTwitterAuthUrl) {
+                setAuthUuid(token);
+                open(
+                    `${backendInMemoryConfig?.airdropTwitterAuthUrl}?tauri=${token}${code ? `&universeReferral=${code}` : ''}`
+                );
+            }
+        },
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [backendInMemoryConfig?.airdropTwitterAuthUrl]
+    );
+
+    const checkAuth = useCallback(() => {
+        if (authUuid && backendInMemoryConfig?.airdropApiUrl) {
+            const interval = setInterval(() => {
+                if (authUuid) {
+                    fetch(`${backendInMemoryConfig?.airdropApiUrl}/auth/twitter/get-token/${authUuid}`, {
+                        method: 'POST',
+                        headers: {
+                            'Content-Type': 'application/json',
+                        },
+                    })
+                        .then((response) => response.json())
+                        .then((data) => {
+                            if (!data.error) {
+                                clearInterval(interval);
+                                setAirdropTokens(data);
+                                restartMining();
+                            }
+                        });
+                }
+            }, 1000);
+            const timeout = setTimeout(
+                () => {
+                    clearInterval(interval);
+                    setAuthUuid('');
+                },
+                1000 * 60 * 5
+            );
+
+            return () => {
```

Comment:
we could probably have this already in a useEffect here and have a `checkAuth` state, so when you do `handleAuth` it sets `setAuthUuid` and `setCheckAuth` to start the interval checking, so usage is contained within the hook itself

Reply:
Good call, just updated it